libsql-server/src/connection/write_proxy.rs

```diff
@@ -332,7 +338,23 @@ where
                         continue;
                     }
 
-                    if !response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?)? {
+                    let should_continue = tokio::task::spawn_blocking({
```

Comment:
This change is related to what @MarinPostma proposed in #775 (I think) - I used it his code, otherwise the tests would deadlock in tests when stream dropping schedules sending close request in background task.